kong/cmd/utils/nginx_signals.lua

```diff
@@ -107,6 +107,25 @@ function _M.start(kong_conf)
   return true
 end
 
+function _M.check_conf(kong_conf)
+  local nginx_bin, err = _M.find_nginx_bin()
+  if not nginx_bin then
+    return nil, err
+  end
+
+  local cmd = fmt("KONG_NGINX_CONF_CHECK=true %s -t -p %s -c %s",
+                  nginx_bin, kong_conf.prefix, "nginx.conf")
+  log.debug("checking nginx conf: %s", cmd)
+
+  local ok, retcode, _, stderr = pl_utils.executeex(cmd)
+  if not ok then
+    return false, ("failed to validate nginx configuration " ..
+           "(exit code %d):\n%s"):format(retcode ,stderr)
+  end
+
```

Comment:
Let's also add a debug log when the validation succeededs, otherwise the user just sees "checking nginx conf" and does not know what happened after that.

Reply:
UNIX philosophy of no output is good?